stdlib/public/core/FloatingPointFromString.swift

```diff
@@ -0,0 +1,2506 @@
+    // Note: division is destructive, overwrites numerator with quotient
+    var remainderNonZero = false
+    divideMPbyMP(work: &work,
+                 numerator: &numeratorRange,
+                 denominator: &denominatorRange,
+                 remainderNonZero: &remainderNonZero)
+    let quotientRange = numeratorRange
+    let quotientBits = bitCountMP(work: work, range: quotientRange)
+    binaryExponent = quotientBits &- 1
+    binaryExponent &+= decimalExponent
+    binaryExponent &-= numeratorShift
+
+    if binaryExponent > targetFormat.minBinaryExponent {
+      // Normal decimal
+      var significand = mostSignificantBitsFrom(work: work,
+                                                range: quotientRange,
+                                                count: targetFormat.significandBits,
+                                                remainderNonZero: remainderNonZero)
+      if significand >= (1 &<< targetFormat.significandBits) {
+        significand >>= 1
+        binaryExponent &+= 1
+      }
+      targetSignificand = significand
+      if binaryExponent > targetFormat.maxBinaryExponent {
+        return .infinity(sign: sign)
+      }
+    } else if binaryExponent >= targetFormat.minBinaryExponent &- targetFormat.significandBits {
+      // Subnormal decimal
+      let bitsNeeded = binaryExponent &- (targetFormat.minBinaryExponent &- targetFormat.significandBits)
+      binaryExponent = targetFormat.minBinaryExponent &- 1 // Flag as subnormal
+      let significand = mostSignificantBitsFrom(work: work,
+                                                range: quotientRange,
+                                                count: bitsNeeded,
+                                                remainderNonZero: remainderNonZero)
+      // Usually, overflowing the expected number of bits doesn't
+      // break anything; it just results in a significand 1 bit longer
+      // than we expected.
+
+      // Except when the significand overflows into the
+      // exponent.  Then we've transitioned from a subnormal to
+      // a normal, so the extra overflow bit will naturally get
+      // dropped, we just have to bump the exponent.
+      if significand >= (1 &<< (targetFormat.significandBits &- 1)) {
+        binaryExponent &+= 1
+      }
+      targetSignificand = significand
+    } else {
+      // Underflow
+      return .zero(sign: sign)
+    }
+  }
+
+  return .binary(significand: targetSignificand,
+                 exponent: Int16(truncatingIfNeeded: binaryExponent),
+                 sign: sign)
+}
+
+// ================================================================
+// ================================================================
+//
+// Float16
+//
+// ================================================================
+// ================================================================
+
+#if !((os(macOS) || targetEnvironment(macCatalyst)) && arch(x86_64))
+
+@available(SwiftStdlib 5.3, *)
```

Comment:
@stephentyrone Any thoughts on whether I can delete this entire function?  Since it was public before, it's conceivable that someone was calling it directly, but no stdlib code ever called it.